arithmetization/src/main/java/net/consensys/linea/zktracer/container/stacked/StackedList.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Consensys Software Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package net.consensys.linea.zktracer.container.stacked;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import lombok.Getter;
+import lombok.experimental.Accessors;
+import net.consensys.linea.zktracer.container.ModuleOperation;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Implements a system of nested lists behaving as a single one, where the current context
+ * modification can transparently be dropped.
+ *
+ * @param <E> the type of elements stored in the list
+ */
+@Accessors(fluent = true)
+public class StackedList<E extends ModuleOperation> {
+  private final List<E> operationSinceBeginningOfTheConflation = new ArrayList<>();
+  @Getter private final List<E> thisTransactionOperation = new ArrayList<>();
```

Comment:
Maybe rename to `operationsInCurrentTransaction`

Reply:
But in all cases the name should start with `operations...`

Reply:
This will also apply to https://github.com/Consensys/linea-tracer/blob/d10cd5ea5dca08e25ccec4477c0ed2cc4ae8b398/arithmetization/src/main/java/net/consensys/linea/zktracer/container/stacked/CountOnlyOperation.java#L21

Reply:
Personal preference: `operationsGeneratedByCurrentTransaction`

Reply:
done